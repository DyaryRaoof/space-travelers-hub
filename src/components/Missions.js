import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { Table, Button } from 'react-bootstrap';
import fetchMissions from '../api';
import { joinMission } from '../redux/missions/missions';
import './Mission.css';

const Missions = () => {
  const missions = useSelector((state) => state.missionsReducer.missions);
  const dispatch = useDispatch();

  useEffect(() => dispatch(fetchMissions()), []);

  return (
    <div className="missions-table">
      <Table striped bordered hover>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>-</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td className="mission-name-cell">{mission.mission_name}</td>
              <td className="mission-description-cell">{mission.description}</td>
              <td className="mission-member-cell">
                <span
                  className={mission.reserved ? 'mission-member-span-active' : 'mission-member-span'}
                >
                  {mission.reserved ? 'Active Member' : 'Not A Member'}
                </span>
              </td>
              <td className="button-cell">
                <Button
                  onClick={() => { dispatch(joinMission(mission.mission_id)); }}
                  variant={mission.reserved ? 'outline-danger' : 'outline-secondary'}
                >
                  {mission.reserved ? 'Leave Mission' : 'Join Mission'}

                </Button>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>

  );
};

export default Missions;
