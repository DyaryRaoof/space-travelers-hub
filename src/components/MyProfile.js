import {
  Container, Col, Row, ListGroup,
} from 'react-bootstrap';
import { useSelector } from 'react-redux';
import './MyProfile.css';

const MyProfile = () => {
  const missions = useSelector((state) => state.missionsReducer
    .missions);

  const rockets = useSelector((state) => state.rocketsReducer
    .rockets);

  return (
    <Container className="profile-container">
      <Row>
        <Col>
          <h1 data-testid="header">My Missions</h1>
          <ListGroup>
            {missions.filter((mission) => mission.reserved === true).map((mission) => (
              <ListGroup.Item key={mission.mission_id}>
                {mission.mission_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </Col>
        <Col>
          <h1>My Rockets</h1>
          <ListGroup>
            {rockets.filter((rocket) => rocket.reserved === true).map((rocket) => (
              <ListGroup.Item key={rocket.id}>
                {rocket.rocket_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </Col>
      </Row>
    </Container>
  );
};

export default MyProfile;
