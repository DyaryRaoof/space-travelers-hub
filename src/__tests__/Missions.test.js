import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import MyProfile from '../components/MyProfile';
import store from '../redux/configureStore';

describe('test Mission component', () => {
  test('should contain table', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    return expect(screen.getByRole('table')).toBeInTheDocument;
  });
});

describe('test with jest snapshot', () => {
  it('should contain heading', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    return expect(screen.getByTestId('header')).toBeInTheDocument;
  });
});

describe('test with jest snapshot', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
