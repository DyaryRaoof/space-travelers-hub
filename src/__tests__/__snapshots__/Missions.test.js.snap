// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with jest snapshot should render correctly 1`] = `
<div
  className="profile-container container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1
        data-testid="header"
      >
        My Missions
      </h1>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="col"
    >
      <h1>
        My Rockets
      </h1>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
  </div>
</div>
`;
